api/handlers/service_instance.go

```diff
@@ -62,6 +64,50 @@ func NewServiceInstance(
 	}
 }
 
+func (h *ServiceInstance) get(r *http.Request) (*routing.Response, error) {
+	authInfo, _ := authorization.InfoFromContext(r.Context())
+	logger := logr.FromContextOrDiscard(r.Context()).WithName("handlers.service-instance.get")
+
+	payload := new(payloads.ServiceInstanceGet)
+	err := h.requestValidator.DecodeAndValidateURLValues(r, payload)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "Unable to decode request query parameters")
+	}
+
+	serviceInstanceGUID := routing.URLParam(r, "guid")
+
+	serviceInstance, err := h.serviceInstanceRepo.GetServiceInstance(r.Context(), authInfo, serviceInstanceGUID)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "failed to get service instance")
+	}
+
+	includedResources, err := h.includeResolver.ResolveIncludes(r.Context(), authInfo, []repositories.ServiceInstanceRecord{serviceInstance}, payload.IncludeResourceRules)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "failed to build included resources")
+	}
+
+	return routing.NewResponse(http.StatusOK).WithBody(presenter.ForServiceInstance(serviceInstance, h.serverURL, includedResources...)), nil
+}
+
+func (h *ServiceInstance) getCredentials(r *http.Request) (*routing.Response, error) {
+	authInfo, _ := authorization.InfoFromContext(r.Context())
+	logger := logr.FromContextOrDiscard(r.Context()).WithName("handlers.service-instance.get-credentials")
+
+	serviceInstanceGUID := routing.URLParam(r, "guid")
+
+	serviceInstance, err := h.serviceInstanceRepo.GetServiceInstance(r.Context(), authInfo, serviceInstanceGUID)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "failed to get service instance")
+	}
+
+	credentials, err := h.serviceInstanceRepo.GetServiceInstanceCredentials(r.Context(), authInfo, serviceInstance.SecretName)
```

Comment:
@danail-branekov I talked about this with @georgethebeatle and we agreed to do it this way, since if we use the service instance guid, we rely on the assumption that the secret has the same name as the instance guid.

Reply:
But you are right, we can use the instance guid, fetch it in the repo and get the secretName from there.

Reply:
The suggestion is to pass the service instance guid so that the repository can lookup the service instance again. This way we do not do any assumptions. We have had troubles in the past with such assumptions and we have already [removed](https://github.com/cloudfoundry/korifi/commit/cd3252c255e17c6d2da441ef5f60f837f8114baf#diff-35ec59eae253edff91b574d395a105f99aba017f6f392c4a50253fecd7bf4b82R213) them. Also, for the theoretical kubectl user this assumption has never existed.

We know it is extra work but we should avoid premature optimisations and this endpoint is not that critical

Reply:
I understand, will do it this way.

Reply:
> what is the final decision here
* Use the instance guid in `serviceInstanceRepo.GetServiceInstanceCredentials`
* Get rid of the `SecretName` field of the `ServiceInstanceRecord`